recipes/qhull/all/conanfile.py

```diff
@@ -1,60 +1,75 @@
 from conan import ConanFile
+from conan.errors import ConanInvalidConfiguration
 from conan.tools.cmake import CMake, CMakeToolchain, cmake_layout
 from conan.tools.files import apply_conandata_patches, copy, export_conandata_patches, get, rmdir
 from conan.tools.microsoft import is_msvc
+from conan.tools.scm import Version
 import os
 
-required_conan_version = ">=1.53.0"
+required_conan_version = ">=2.0.9"
 
 
 class QhullConan(ConanFile):
     name = "qhull"
-    description = "Qhull computes the convex hull, Delaunay triangulation, " \
-                  "Voronoi diagram, halfspace intersection about a point, " \
-                  "furthest-site Delaunay triangulation, and furthest-site " \
-                  "Voronoi diagram."
+    description = ("Qhull computes the convex hull, Delaunay triangulation, "
+                   "Voronoi diagram, halfspace intersection about a point, "
+                   "furthest-site Delaunay triangulation, and furthest-site "
+                   "Voronoi diagram.")
     license = "Qhull"
-    topics = ("geometry", "convex", "triangulation", "intersection")
-    homepage = "http://www.qhull.org"
     url = "https://github.com/conan-io/conan-center-index"
-
+    homepage = "http://www.qhull.org"
+    topics = ("geometry", "convex", "triangulation", "intersection")
     package_type = "library"
     settings = "os", "arch", "compiler", "build_type"
     options = {
         "shared": [True, False],
         "fPIC": [True, False],
+        "cpp": [True, False],
         "reentrant": [True, False],
     }
     default_options = {
         "shared": False,
         "fPIC": True,
+        "cpp": False,
```

Comment:
The upstream does not really support an option for that, instead, users can decide what they will consume based on cmake targets. I would suggest removing `cpp` option, it only increases recipe complexity and is not clear about the benefit, as there is no related issue.

Reply:
it does need that logic, again, only increases complexity. It could be decided in `package_info` if is installed or not. Besides that, it's an alpha version (long period, but still alpha) and may change.

Reply:
Anyway, if you meant dropping the cpp component entirely, then I'm definitely against that. There's even an unrelated issue requesting support for it.

Reply:
> I would actually enable the cpp component by default

Agreed.